frontend/app.py

```diff
@@ -575,39 +592,43 @@ async def download(
     return create_download_response(download_request, canarydrop=canarydrop)
 
 
+@app.get("/commitsha")
+def get_commit_sha():
+    commit_sha = get_deployed_commit_sha()
+    return JSONResponse({"commit_sha": commit_sha}, status_code=200)
+
+
+# remove up until here once new ui stable
+
+
+# NOTE: Do not remove this when cleaning up after UI is stable
 @app.get("/azure_css_landing", tags=["Azure Portal Phishing Protection App"])
 async def azure_css_landing(
-    request: Request, admin_consent: str = "", tenant: str = None, state: str = None
+    request: Request,
+    admin_consent: str = "",
+    tenant: str = None,
+    state: str = None,
+    error: str = None,
 ) -> HTMLResponse:
     """
     This page is loaded after a user has authN and authZ'd into their tenant and granted the permissions to install the CSS
     Once the CSS is installed into their tenant, and we revoke our permission grants, we can close the window as this will happen in
     a pop-up context.
     """
-    info = ""
-    if admin_consent == "True":
-        css = None
-        token = None
-        token_auth = None
-
-        encodedCss, token, token_auth = b64decode(unquote(state)).decode().split(":")
-        css = b64decode(encodedCss).decode()
-
-        if css is not None and tenant is not None:
-            (success, info) = install_azure_css(tenant, css)
-            info += " We have uninstalled our application from you tenant, revoking all of our permissions."
+    css = b64decode(unquote(state)).decode()
+
+    if not admin_consent == "True" or error == EntraTokenErrorAccessDenied:
+        status = ENTRA_STATUS_NO_ADMIN_CONSENT
     else:
-        info = "Installation failed due to lack of sufficient granted permissions."
-    return templates.TemplateResponse(
-        "azure_install.html",
-        {"request": request, "status": info},
-    )
+        status = install_azure_css(tenant, css)
 
+    if not frontend_settings.NEW_UI:
+        return templates.TemplateResponse(
+            "azure_install.html",
+            {"request": request, "status": LEGACY_ENTRA_STATUS_MAP[status]},
```

Comment:
Consider returning for example:

```
"status":  "has_custom_css_already'
"status_msg": "Installation failed: your .... "
```

Then the frontend (or other system using the API) has a machine-checkable status if it needs to make a decision based on it, and message if it has to render one for user.